crates/telemetry-subscribers/src/lib.rs

```diff
@@ -367,7 +390,13 @@ impl TelemetryConfig {
 
         // The guard must be returned and kept in the main fn of the app, as when it's dropped then the output
         // gets flushed and closed. If this is dropped too early then no output will appear!
-        let guards = TelemetryGuards { worker_guard };
+        let mut guards = TelemetryGuards { worker_guard };
+        if config.gas_stats_filter_prefix.is_some() {
+            // if the gas stats filtering is enabled the provided file will only be used to record gas stats
+            guards = TelemetryGuards {
+                worker_guard: gas_guard,
+            }
+        }
```

Comment:
The tracing infrastructure is very flexible and allows for having a "stack" of "layers". This means that instead of disabling the normal logging, you can add another parallel "layer" that handles and filters for the tracing events you care about and handles them appropriately, all with keeping intact the normal logging

Reply:
Ok, so sounds like we need to keep both the handles and return both in a vector?

Reply:
It will probably take a little bit more work than just returning both in a vector, since you're using the same file handle for both pipes they'll end up colliding.

Reply:
I'll use a separate file then, the reason I did it this was was that I only saw a file being set in the config in one location which was not in the sui-node main.rs instantiation, but I guess that doesn't matter